### PR TITLE
add PooledBuffer to dedup cache key

### PR DESCRIPTION
**Pull request by matthiasdiener**
Followup of #302.

### CODE REVIEW

**Comment by kaushikcfd** [comment]
I wonder if we should check for `pyopencl._cl.MemoryObjectHolder` instead of `(CLBuffer, PooledCLBuffer)`.

File: doc/conf.py [hunk 1]
```diff
@@ -35,5 +35,5 @@
 nitpick_ignore_regex = [
     ["py:class", r"numpy.(u?)int[\d]+"],
     ["py:class", r"pyrsistent.typing.(.+)"],
-
```
Comment (matthiasdiener) [inline doc/conf.py:38]: Is there a better way to fix this doc build error:

```
/home/runner/work/pytato/pytato/.miniforge3/envs/testing/lib/python3.10/site-packages/pytools/tag.py:docstring of pytools.tag.Taggable.tagged:: WARNING: py:class reference target not found: typing_extensions.Self

/home/runner/work/pytato/pytato/.miniforge3/envs/testing/lib/python3.10/site-packages/pytools/tag.py:docstring of pytools.tag.Taggable.without_tags:: WARNING: py:class reference target not found: typing_extensions.Self
```

**Comment by matthiasdiener** [comment]
> I wonder if we should check for `pyopencl._cl.MemoryObjectHolder` instead of `(CLBuffer, PooledCLBuffer)`. (@inducer might have a better handle at this)

Done in [b0023d2](https://github.com/inducer/pytato/pull/304/commits/b0023d2666b0a0bfd35ff13d8e7d247710867fc4)

File: doc/conf.py (hunk 1 above)
Comment (inducer) [reply doc/conf.py:38]: Not that I'm aware:

https://github.com/inducer/pytools/blob/9a21c88c241da64407e78141977870df9ea2d64c/doc/conf.py#L36

**Comment by inducer** [comment]
Thanks!

**Pull request merged**